### PR TITLE
tweaks to build with java21 and remove finalize()

### PR DESCRIPTION
**Pull request by ronreynolds**
specific to Java-21 build (also tested with Java-11 locally):
* upgrade gradle to 8.9
* upgrade mockito to work with Java-21 bytecodes
also
* removed `SmartsheetImpl.finalize()` as it's deprecated and has GC impact due to requiring 2 GC passes to actually delete an object.
  * added `CleanerUtil` to prevent this-capture in this code (javadocs in class explain what that means)   
* bumped the method-count to reduce check-style noise.
* removed the body assertions from LoggingIT since they're no longer logged

### CODE REVIEW

**Comment by ronreynolds** [comment]
@zromano / @apederson94  i had originally intended this for `release/4.0` but it should be safe for `mainline`.  happy to update it to merge into `release/4.0` instead if we want to avoid any risk.  also need to create a `CleanerUtilTest` just to be absolutely sure the `CleanerUtil` code is working as intended.

**Comment by ronreynolds** [comment]
`CleanerUtilTest` added and confirmed it works as intended.

**Review by priyankalakhe — DISMISSED** [review]

**Comment by ronreynolds** [comment]
@priyankalakhe unfortunately the checkstyle rules are a bit fiddly and i had to push some fresh tweaks (with no functional changes whatsoever). :-/ (and it looks like this repo is configured to reset approvals after any new push which makes total sense but is still rather inconvenient).

**Review by priyankalakhe — APPROVED** [review]

File: src/main/java/com/smartsheet/api/internal/util/CleanerUtil.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+     * @param cleaningAction - code to execute when the objToWatch becomes "phantom reachable" (i.e., about to be GCed)
+     * @return a Cleanable which can be used to force a clean() action (and unregisters the objToWatch-cleaningAction)
+     */
+    public static Cleaner.Cleanable register(@NotNull Object objToWatch, @NotNull Runnable cleaningAction) {
+        return CLEANER.register(objToWatch, cleaningAction);
+    }
+
+    /**
+     * this method takes a ref value and returns a Runnable that, when invoked, will close it.  this method actually
+     * makes the issue of ref-capture mentioned above impossible (since the Closeable ref is copied into the Runnable
```
Comment (ronreynolds) [inline src/main/java/com/smartsheet/api/internal/util/CleanerUtil.java:51]: not quite impossible; if the Closeable has a ref to the objectToWatch then you have a circular ref chain which will create an immortal object.  that is unlikely but not impossible and i'm not sure there's a way to solve that. :-/

**Pull request merged**